File: Basic/Check if strings are rotations of each other or not.cs
// { Driver Code Starts
//Initial Template for C#



using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DriverCode
{

    class GFG
    {
        static void Main(string[] args)
        {
            int testcases;// Taking testcase as input
            testcases = Convert.ToInt32(Console.ReadLine());
            while (testcases-- > 0)// Looping through all testcases
            {
                string a = Console.ReadLine().Trim();
                string b = Console.ReadLine().Trim();
                Solution obj = new Solution();
                bool res = obj.areRotations(a,b);
                if(res){
                    Console.Write("1");
                }
                else{
                    Console.Write("0");
                }
                Console.Write("\n");
          }

        }
    }
    
    
}
// } Driver Code Ends


//User function Template for C#


class Solution
    {
        //Complete this function
        public bool areRotations(string a, string b)
        {
            //Your code here
            
            if (a.Length != b.Length) return false;
            
            if (a==b) return true;
            
            for (int i=0; i<a.Length; i++)
            {
                string ns = a;
                ns = a.Substring(1,a.Length-1) + a.Substring(0,1);
                a = ns;
                if (a==b) return true;
            }
           
           return false;
           
           
        }

    }

File: Basic/Check if strings are rotations of each other or not.py
#User function Template for python3

class Solution:
    
    #Function to check if two strings are rotations of each other or not.
    def areRotations(self, s1, s2):
        #code here
        n = len(s1)
        #print(n, "\n")
        
        if n != len(s2):
            return 0
            
        for i in range(n):
            ok = True
            
            for j in range(n):
                pos1 = j
                primo = s1[pos1]
                
                pos2 = (i + j) % n
                secondo = s2[pos2]
                
                #print(pos1, pos2)
                
                if primo != secondo:
                    ok = False
                    break
            
            if ok:
                return 1
                
        
        return 0

#{ 
 # Driver Code Starts
#Initial Template for Python 3
import atexit
import io
import sys

_INPUT_LINES = sys.stdin.read().splitlines()
input = iter(_INPUT_LINES).__next__
_OUTPUT_BUFFER = io.StringIO()
sys.stdout = _OUTPUT_BUFFER

@atexit.register

def write():
    sys.__stdout__.write(_OUTPUT_BUFFER.getvalue())

if __name__=='__main__':
    t = int(input())
    for i in range(t):
        s1=str(input())
        s2=str(input())
        if(Solution().areRotations(s1,s2)):
            print(1)
        else:
            print(0)
    
        
# } Driver Code Ends
